lesson_16/objects/objects_app/src/main/java/com/codedifferently/lesson16/SmartPhone/SmartPhone.java

```diff
@@ -0,0 +1,61 @@
+package com.codedifferently.lesson16.SmartPhone;
+
+import java.util.ArrayList;
+
+public class SmartPhone {
+
+  private Name brandName;
+  private String model;
+  private int batteryPercentage;
+  private boolean power;
+  private ArrayList<String> apps;
+  private int volume;
+
+  public SmartPhone(
+      Name brandName, String model, int batteryPercentage, boolean power, ArrayList<String> apps) {
```

Comment:
Should be `List<String> apps` so that clients can pass in any implementation of a `List` that works for them. This meets the spirit of the dependency inversion principle.